.github/workflows/reusable-set-version.yml

```diff
@@ -0,0 +1,122 @@
+        run: |
+          echo "PR_TITLE=$PR_TITLE"
+          RELEASE_TYPE="invalid"
+          echo "$PR_TITLE" | grep "\[patch\]" > /dev/null && export RELEASE_TYPE=patch
+          echo "$PR_TITLE" | grep "\[minor\]" > /dev/null && export RELEASE_TYPE=minor
+          echo "$PR_TITLE" | grep "\[major\]" > /dev/null && export RELEASE_TYPE=major
+          echo "RELEASE_TYPE=$RELEASE_TYPE"
+          if [ "$RELEASE_TYPE" == "invalid" ]; then
+            echo "PR title shall contain [patch] or [minor] or [major], but got: $PR_TITLE"
+            exit 1;
+          fi
+          echo "RELEASE_TYPE=$RELEASE_TYPE" >> $GITHUB_ENV
+      - name: Checkout sources
+        if: github.event_name == 'push'
+        uses: actions/checkout@v4
+        with:
+          submodules: 'false'
+          fetch-depth: 0
+      - name: Compute current version
+        if: github.event_name == 'push'
+        id: set_version
+        continue-on-error: true
+        run: |
+          git describe --tags # git describe fails if there are no tags yet. Required to fail the step.
+          LATEST_TAG=`git describe --tags | cut -d'-' -f1 | tr -d v`
+          echo "LATEST_TAG=$LATEST_TAG"
+          MAJOR=`echo $LATEST_TAG | cut -d'.' -f1`
+          MINOR=`echo $LATEST_TAG | cut -d'.' -f2`
+          PATCH=`echo $LATEST_TAG | cut -d'.' -f3`
+          echo "MAJOR=$MAJOR" >> $GITHUB_ENV
+          echo "MINOR=$MINOR" >> $GITHUB_ENV
+          echo "PATCH=$PATCH" >> $GITHUB_ENV
+      - name: Fallback set version
+        if: steps.set_version.outcome != 'success' && github.event_name == 'push'
+        id: fallback
+        run: |
+          echo "No last tag detected, assuming 0.0.0"
+          echo "MAJOR=0" >> $GITHUB_ENV
+          echo "MINOR=0" >> $GITHUB_ENV
+          echo "PATCH=0" >> $GITHUB_ENV
+      - name: Version increment
+        if: github.event_name == 'push'
+        id: inc_version
+        run: |
+          case $RELEASE_TYPE in
+            major)
+              MAJOR=$((MAJOR+1))
+              MINOR=0
+              PATCH=0
+              ;;
+            minor)
+              MINOR=$((MINOR+1))
+              PATCH=0
+              ;;
+            patch)
+              PATCH=$((PATCH+1))
+              ;;
+            *)
+              echo "Unknown release type"
+              exit 1
+              ;;
+          esac
+          TAG=$MAJOR.$MINOR.$PATCH
+          if git rev-parse -q --verify "refs/tags/$TAG" >/dev/null; then
+            echo "tag $TAG already exists, aborting. To fix this, you need manually delete tag $TAG"
+            exit 1
+          fi
+
+          echo "VERSION=$TAG"
+          echo "VERSION=$TAG" >> $GITHUB_ENV
+          echo "VERSION=$TAG" >> $GITHUB_OUTPUT
+          echo "$TAG" > .VERSION
+
+      - name: Write version to file
+        if: github.event_name == 'push'
+        uses: brettdorrans/write-version-to-file@master
+        with:
+          filename: '.VERSION'
+      - name: Commit VERSION file
```

Comment:
This will break as soon as we introduce signed commits. Automation won't be permitted to create commits at all.